tests/ISA/ISA.Tests/CompositeHeader.Tests.fs

```diff
@@ -26,11 +26,19 @@ let private tests_iotype =
             Expect.equal (IOType.FreeText "Test").asOutput "Output [Test]" "FreeText Test"
         )
         // This test ensures that new IOTypes are also added to `All` static member.
-        testCase "All" (fun () -> 
-            let caseInfos = Microsoft.FSharp.Reflection.FSharpType.GetUnionCases(typeof<IOType>) 
-            let count = caseInfos |> Array.length
-            Expect.hasLength IOType.All (count-1) "Expect one less than all because we do not want to track `FreeText` case."
+        testCase "Cases" (fun () -> 
+            let caseInfos = IOType.Cases
+            Expect.hasLength IOType.All (caseInfos.Length-1) "Expect one less than all because we do not want to track `FreeText` case."
         )
+        testCase "getExplanation" <| fun _ ->
+            let cases = IOType.Cases |> Array.map snd
+            for case in cases do
+                let actual = IOType.getExplanation(U2.Case2 case)
```

Comment:
Seems a bit weird in F#. Any other workaround?

Reply:
`getExplanation`/`GetExplanation` is a pure UI feature. Its design aims to prove a iterative method for showing all cases and respective explanation. 

If you dislike the design, we can move it to a more hidden `UI` module without flooding our main API space.

Reply:
If it's really exclusively meant for the UI, how about adjusting the naming to accomodate this? E.g. "getUIToolTip" or smt like this